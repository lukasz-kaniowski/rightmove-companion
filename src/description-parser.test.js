const parser = require('./description-parser');
const sampleText = `
Some other text
Lounge 12' 3" x 13' 11" (3.73m x 4.24m )
Kitchen / Dining Room 17' 3" x 8' 3" (5.26m x 2.51m )
Ground Floor Bathroom 6' 0" x 5' 7" (1.83m x 1.7m )
Bedroom 15' 3" x 11' 9" (4.65m x 3.58m )
Bedroom 12' 7" x 8' 5" (3.84m x 2.57m )
Bedroom 8' 4" x 6' 5" (2.54m x 1.96m )
Double glazed window to rear, radiator, laid to carpet
Garden 11' 0" x 16' 0" (3.35m x 4.88m )`;

describe('description parser', () => {
  let result;
  beforeEach(() => {
    result = parser.parse(sampleText)
  });

  it('should find all rows with dimensions', () => {
    expect(result.items.length).toBe(7)
  });

  it('should return correct measures', () => {
    expect(result.items).toMatchSnapshot();
  });

  it('should return correct sum', () => {
    expect(result.sum).toBe(79.971)
  });
});
