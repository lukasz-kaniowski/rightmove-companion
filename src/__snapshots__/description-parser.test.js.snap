// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`description parser should return correct measures 1`] = `
Array [
  Object {
    "a": "3.73",
    "b": "4.24",
    "name": "Lounge",
    "size": 15.8152,
  },
  Object {
    "a": "5.26",
    "b": "2.51",
    "name": "Kitchen",
    "size": 13.202599999999999,
  },
  Object {
    "a": "1.83",
    "b": "1.7",
    "name": "Ground Floor",
    "size": 3.111,
  },
  Object {
    "a": "4.65",
    "b": "3.58",
    "name": "Bedroom",
    "size": 16.647000000000002,
  },
  Object {
    "a": "3.84",
    "b": "2.57",
    "name": "Bedroom",
    "size": 9.868799999999998,
  },
  Object {
    "a": "2.54",
    "b": "1.96",
    "name": "Bedroom",
    "size": 4.9784,
  },
  Object {
    "a": "3.35",
    "b": "4.88",
    "name": "Garden",
    "size": 16.348,
  },
]
`;
